bin/prover-client/src/proving_ops/cl_ops.rs

```diff
@@ -19,14 +20,20 @@ use crate::{
 pub struct ClOperations {
     cl_client: HttpClient,
     el_dispatcher: Arc<TaskDispatcher<ElOperations>>,
+    rollup_params: Arc<RollupParams>,
 }
 
 impl ClOperations {
     /// Creates a new CL operations instance.
-    pub fn new(cl_client: HttpClient, el_dispatcher: Arc<TaskDispatcher<ElOperations>>) -> Self {
+    pub fn new(
+        cl_client: HttpClient,
+        el_dispatcher: Arc<TaskDispatcher<ElOperations>>,
+        rollup_params: Arc<RollupParams>,
```

Comment:
Should be `Arc<Params>`, we don't pass around the inner `RollupParams` in an arg.  Although this might all get refactored soon anyways.

Reply:
Why shouldn't we pass `RollupParams`? In the `prover-client`, non-rollup params are not required.

Reply:
Why not just a `&RollupParams` and let the user `Arc`it if necessary?

Reply:
Or `&Params` just don't shove an `Arc` down the throat.

Reply:
The goal was to only have one instance of the `Params` type and its constitutents, so that's why we'd be passing it around as an `Arc`.